### PR TITLE
feat: Rework LoginContext to Redux store

### LINKED ISSUE #379: Rework LoginContext to store

### PR DESCRIPTION
**Pull request by dmutrokot**
Resolves #379

### CODE REVIEW

File: src/core/store/login/slices/login.slice.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+import { LoginAddress } from 'shared/models';
+
+export interface LoginState {
+    loggedAddresses: LoginAddress[];
+    activeAddress: string | null | undefined;
```
Comment (Vladislav94) [inline src/core/store/login/slices/login.slice.ts:7]: can it be just one type?

Comment (dmutrokot) [reply src/core/store/login/slices/login.slice.ts:7]: unfortunately no, `null` goes from local storage and `undefined` from url param

File: src/components/Login/LoginButton.tsx [hunk 2]
```diff
@@ -44,45 +44,55 @@ export function LoginButton() {
...
+                name: address.slice(0, 6)
+            }));
         }
     };
 
     return (
         <>
-            <div className={classNames(classes.button, dropdownOpen ? 'opened' : 'closed')}>
-                <div className={classes.buttonInner} onClick={dropdownToggle}>
+            <div className={classNames(classes.button, isDropdownOpen ? 'opened' : 'closed')}>
```
[the remark this answers is not in the document]
Comment (dmutrokot) [reply src/components/Login/LoginButton.tsx]: fixed, thanks

File: src/components/Login/LoginButton.tsx [hunk 3]
```diff
@@ -129,8 +139,8 @@ export function LoginButton() {
 
             <Backdrop
                 sx={{ }}
```
[the remark this answers is not in the document]
Comment (dmutrokot) [reply src/components/Login/LoginButton.tsx]: fixed, thanks

**Review by dudendy — APPROVED** [review]

**Pull request merged**